### PR TITLE
Update OpenMS to 3.1.0

### PR DESCRIPTION
**Pull request by timosachsenberg**
Update the OpenMS and pyOpenMS recipes to the 3.1.0 release

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/openms-meta/meta.yaml [hunk 1]
```diff
@@ -11,12 +11,14 @@ source:
...
-  url: https://abibuilder.cs.uni-tuebingen.de/archive/openms/OpenMSInstaller/release/{{ version }}/OpenMS-{{ version }}-src.tar.gz
-  md5: 2b7ee0f4d8f42eb3c17384347df2dba4  
+  url: https://github.com/OpenMS/OpenMS/releases/download/Release{{ version }}/OpenMS-{{ version }}-src.tar.gz
+  md5: 6a60ef1a95c7ce915cc3a02e5bccd593
 {% endif %}
 
 build:
   number: 1
+  run_exports:
+    - {{ pin_subpackage({{ name }}, max_pin="x.x") }}
```
Comment (timosachsenberg) [inline recipes/openms-meta/meta.yaml:21]:
```suggestion
    - {{ pin_subpackage(name, max_pin="x.x") }}  
```

File: recipes/openms-meta/meta.yaml [hunk 2]
```diff
@@ -11,12 +11,14 @@ source:
...
-  url: https://abibuilder.cs.uni-tuebingen.de/archive/openms/OpenMSInstaller/release/{{ version }}/OpenMS-{{ version }}-src.tar.gz
-  md5: 2b7ee0f4d8f42eb3c17384347df2dba4  
+  url: https://github.com/OpenMS/OpenMS/releases/download/Release{{ version }}/OpenMS-{{ version }}-src.tar.gz
+  md5: 6a60ef1a95c7ce915cc3a02e5bccd593
 {% endif %}
 
 build:
   number: 1
+  run_exports:
+    - {{ pin_subpackage('openms-meta', max_pin="x.x") }}
```
Comment (timosachsenberg) [inline recipes/openms-meta/meta.yaml:21]:
```suggestion
    - {{ pin_subpackage('openms', max_pin="x.x") }}  
```

**Comment by timosachsenberg** [comment]
Rerunning this for the second time now but still get: https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=44154&view=results
"[error]We stopped hearing from agent Hosted Agent. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610"

@bgruening do you know by chance how to fix/debug this?

**Comment by timosachsenberg** [comment]
pinging @bernt-matthias do you know what could be wrong here?

File: recipes/openms-meta/build.sh [hunk 3]
```diff
@@ -46,6 +46,6 @@ cmake .. \
   -DBUILD_EXAMPLES=OFF
 
 # limit concurrent build jobs due to memory usage on CI
-make -j4 OpenMS TOPP UTILS
+make -j4 OpenMS TOPP
```
Comment (timosachsenberg) [inline recipes/openms-meta/build.sh:49]:
```suggestion
make -j2 OpenMS TOPP
```

**Comment by timosachsenberg** [comment]
maybe an out of memory issue... with 2 threads it seems to run now. Let's hope it doesn't get killed

**Comment by timosachsenberg** [comment]
ok went through!

**Comment by timosachsenberg** [comment]
@BiocondaBot please add label

File: recipes/openms-meta/meta.yaml [hunk 4]
```diff
@@ -143,14 +145,14 @@ outputs:
...
        - comet-ms ==2023010
-       - fido ==1.0
        - luciphor2 ==2020_04_03
-       - msgf_plus ==2021.03.22
+       - msgf_plus ==2023.01.1202
        - pepnovo ==20101117
        - percolator ==3.5
-       - sirius-csifingerid =4.9
-       - thermorawfileparser ==1.3.4
+       - sirius-ms ==5.8.2 # from conda-forge
```
Comment (bernt-matthias) [inline recipes/openms-meta/meta.yaml:152]: I remember that there was an explicit way to refer to a package from a specific channel.

Comment (bernt-matthias) [reply recipes/openms-meta/meta.yaml:152]:
```suggestion
       - conda-forge::sirius-ms ==5.8.2
```

Comment (timosachsenberg) [reply recipes/openms-meta/meta.yaml:152]: ah good to know. I am fine with that if it works. Or should I change something?

**Review by ypriverol — APPROVED** [review]

**Comment by bgruening** [comment]
I think the VMs only have 2 cores. Glad you solved it.

File: recipes/openms-meta/meta.yaml (hunk 4 above)
Comment (bgruening) [reply recipes/openms-meta/meta.yaml]: There is only one from conda-forge? Why is this needed?
Also please keep in mind that we prioritise conda-forge over bioconda if a package is in both channels.

**Comment by jpfeuffer** [comment]
Why is there no log output anymore?
That is probably also the reason that jobs start failing due to non-existent output for minutes

**Comment by timosachsenberg** [comment]
@BiocondaBot please add label

**Review by ypriverol — APPROVED** [review]

**Comment by timosachsenberg** [comment]
@bgruening ready to go

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @timosachsenberg and Co! Very nice release!

**Comment by timosachsenberg** [comment]
Thanks 🙏

**Comment by ypriverol** [comment]
Thanks @bgruening

**Pull request merged**